### PR TITLE
Varnish ESI Block

### PR DESCRIPTION
**Pull request by goivvy**
## Purpose of this pull request

This pull request (PR) explains how to use ESI with Varnish on Magento 2

Fixes https://github.com/magento/devdocs/issues/9134

### CODE REVIEW

File: src/guides/v2.4/config-guide/varnish/use-varnish-esi.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+An example:
+
+```html
+<div>
+  <esi:include src="http://domain.com/index.php/page_cache/block/esi/blocks"/>
+</div>
+```
+
+Varnish will fetch content from `http://domain.com/index.php/page_cache/block/esi/blocks` and put it in place of `esi` tag.
```
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/varnish/use-varnish-esi.md:18]:
```suggestion
Varnish will fetch content from `http://domain.com/index.php/page_cache/block/esi/blocks` and put it in place of the `esi` tag.
```

File: src/guides/v2.4/config-guide/varnish/use-varnish-esi.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+`cms_index_index.xml`:
+
+```xml
+  <referenceContainer name="content">
+      <block class="Magento\Framework\View\Element\Template" template="Magento_Paypal::esi.phtml" ttl="30"/>
+   </referenceContainer>
+```
+
+In the example above content from `esi.phtml` will be added to a homepage and will be automatically updated by Varnish every 30 seconds.
```
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/varnish/use-varnish-esi.md:37]:
```suggestion
In the example above, content from `esi.phtml` will be added to a homepage and will be automatically updated by Varnish every 30 seconds.
```

File: src/guides/v2.4/config-guide/varnish/use-varnish-esi.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+## Limitations
+
+Currently, Varnish doesn't support ESI over https so Magento 2 will automatically switch to http.
```
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/varnish/use-varnish-esi.md:41]: Please don't use contractions
```suggestion
Currently, Varnish does not support ESI over HTTPS so Magento 2 will automatically switch to HTTP.
```

File: src/guides/v2.4/config-guide/varnish/use-varnish-esi.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+`Magento\PageCache\Observer\ProcessLayoutRenderElement`:
+
+```php
+    private function _wrapEsi(
+        \Magento\Framework\View\Element\AbstractBlock $block,
+        \Magento\Framework\View\Layout $layout
+    ) {
+    ....
+        // Varnish does not support ESI over HTTPS must change to HTTP
```
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/varnish/use-varnish-esi.md:51]:
```suggestion
        // Varnish does not support ESI over HTTPS, must change to HTTP
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Great new topic! Just a few small grammar issues to correct.

**Comment by goivvy** [comment]
@BarnyShergold corrected.

**Review by BarnyShergold — APPROVED** [review]
Excellent

**Comment by dobooth** [comment]
Thanks for the excellent contribution, @goivvy. One more thing if you would. We generally symlink upwards: 2.3 as content, 2.4 symlinked. Can you please reverse your topic/symlink?

**Comment by goivvy** [comment]
@dobooth @BarnyShergold fixed.

**Review by dobooth — APPROVED** [review]

**Comment by dobooth** [comment]
@magento import code to magento-devdocs/devdocs

**Comment by m2-github-services** [comment]
@dobooth the branch with the code is successfully imported into the `magento-devdocs/devdocs` repository. Branch name: `imported-goivvy-devdocs-9380`.

**Pull request merged**